Introduce Resize Factors for map font/lines/symbols

### Tell us your idea!

In OSM map elements, esp. text is quite hard to read (i. e. really small) on high resolution displays, esp. if you use Elevate5 as map theme.  Therefore, I have written a script which increases fonts, lines and symbols by a configurable factor, which massively increases usability.

The other day I learned two things:
1. I am not the only one with this "problem". 
2. Locus has a setting for this, in fact, to increase all map elements by a configurable factor ...

... what gave me the idea that c:geo could perhaps offer this functionality, too.  ;)

### What solution would you suggest?

From my experience, it turned out to be best to have three factors:

1. Font Size
2. Line Width
3. Symbol Size

... as in fact I like the theme best, if fonts are increased to 160, lines to 120 and symbols to 130 percent,
while my teammate prefers F140/L120/S110 and a friend of ours even wanted to have F180/L120/S130.
Although my script can create all three in one go, life would be way easier, if c:geo would support to do this itself.  ;)

### Is there an alternative?

Just write yourself a script to increase sizes ...  ;)

### Additional Information

As an example here a comparison between original (in fact close to unreadable) vs. "my" F160, L120 S130:

![Current](https://user-images.githubusercontent.com/9889015/166129743-99b5491b-2d88-42ca-9475-952389318942.jpg) ![Font 160 Lines 120 Symbols 130](https://user-images.githubusercontent.com/9889015/166129750-7206c548-d96b-436b-b068-17b710c58b58.jpg)

Generally works fine now with Freizeitkarte theme, thanks for adapting the code.

While doing my quick test I got a few exceptions, though. Not yet sure whether they are reproducible (and related to scaling at all).
I'd set symbol scaling to "10.0" and received this OOM:

```
2022-05-01 20:02:29.565 7472-7472/cgeo.geocaching E/art: Throwing OutOfMemoryError "Failed to allocate a 28236 byte allocation with 17256 free bytes and 16KB until OOM"
2022-05-01 20:02:29.611 7472-7472/cgeo.geocaching E/AndroidRuntime: FATAL EXCEPTION: main
    Process: cgeo.geocaching, PID: 7472
    java.lang.OutOfMemoryError: Failed to allocate a 28236 byte allocation with 17256 free bytes and 16KB until OOM
        at dalvik.system.VMRuntime.newNonMovableArray(Native Method)
        at android.graphics.BitmapFactory.nativeDecodeAsset(Native Method)
        at android.graphics.BitmapFactory.decodeStream(BitmapFactory.java:609)
        at android.graphics.BitmapFactory.decodeResourceStream(BitmapFactory.java:444)
        at android.graphics.BitmapFactory.decodeResource(BitmapFactory.java:467)
        at android.graphics.BitmapFactory.decodeResource(BitmapFactory.java:497)
        at android.graphics.drawable.BitmapDrawable.updateStateFromTypedArray(BitmapDrawable.java:742)
        at android.graphics.drawable.BitmapDrawable.inflate(BitmapDrawable.java:709)
        at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:1095)
        at androidx.appcompat.graphics.drawable.AnimatedStateListDrawableCompat.parseItem(AnimatedStateListDrawableCompat.java:591)
        at androidx.appcompat.graphics.drawable.AnimatedStateListDrawableCompat.inflateChildElements(AnimatedStateListDrawableCompat.java:495)
        at androidx.appcompat.graphics.drawable.AnimatedStateListDrawableCompat.inflate(AnimatedStateListDrawableCompat.java:208)
        at androidx.appcompat.graphics.drawable.AnimatedStateListDrawableCompat.createFromXmlInner(AnimatedStateListDrawableCompat.java:177)
        at androidx.appcompat.widget.ResourceManagerInternal$AsldcInflateDelegate.createFromXmlInner(ResourceManagerInternal.java:545)
        at androidx.appcompat.widget.ResourceManagerInternal.loadDrawableFromDelegates(ResourceManagerInternal.java:290)
        at androidx.appcompat.widget.ResourceManagerInternal.getDrawable(ResourceManagerInternal.java:142)
        at androidx.appcompat.widget.ResourceManagerInternal.getDrawable(ResourceManagerInternal.java:135)
        at androidx.appcompat.content.res.AppCompatResources.getDrawable(AppCompatResources.java:104)
        at androidx.appcompat.widget.AppCompatCompoundButtonHelper.loadFromAttributes(AppCompatCompoundButtonHelper.java:76)
        at androidx.appcompat.widget.AppCompatCheckBox.<init>(AppCompatCheckBox.java:76)
        at com.google.android.material.checkbox.MaterialCheckBox.<init>(MaterialCheckBox.java:65)
        at com.google.android.material.checkbox.MaterialCheckBox.<init>(MaterialCheckBox.java:61)
        at com.google.android.material.theme.MaterialComponentsViewInflater.createCheckBox(MaterialComponentsViewInflater.java:49)
        at androidx.appcompat.app.AppCompatViewInflater.createView(AppCompatViewInflater.java:139)
        at androidx.appcompat.app.AppCompatDelegateImpl.createView(AppCompatDelegateImpl.java:1563)
        at androidx.appcompat.app.AppCompatDelegateImpl.onCreateView(AppCompatDelegateImpl.java:1614)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:725)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:482)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:414)
        at androidx.appcompat.view.menu.ListMenuItemView.insertCheckBox(ListMenuItemView.java:311)
        at androidx.appcompat.view.menu.ListMenuItemView.setCheckable(ListMenuItemView.java:188)
        at androidx.appcompat.view.menu.ListMenuItemView.initialize(ListMenuItemView.java:129)
        at androidx.appcompat.view.menu.MenuAdapter.getView(MenuAdapter.java:109)
        at androidx.appcompat.view.menu.MenuPopup.measureIndividualMenuWidth(MenuPopup.java:161)
        at androidx.appcompat.view.menu.StandardMenuPopup.tryShow(StandardMenuPopup.java:174)
        at androidx.appcompat.view.menu.StandardMenuPopup.show(StandardMenuPopup.java:208)
        at androidx.appcompat.view.menu.MenuPopupHelper.showPopup(MenuPopupHelper.java:296)
        at androidx.appcompat.view.menu.MenuPopupHelper.tryShow(MenuPopupHelper.java:182)
        at androidx.appcompat.view.menu.MenuPopupHelper.show(MenuPopupHelper.java:144)
        at androidx.appcompat.widget.ActionMenuPresenter.onSubMenuSelected(ActionMenuPresenter.java:310)
        at androidx.appcompat.view.menu.MenuBuilder.dispatchSubMenuSelected(MenuBuilder.java:314)
        at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:1008)
    	at androidx.appcompat.view.menu
2022-05-01 20:02:33.934 1543-1666/system_process E/InputDispatcher: channel 'dcd4654 cgeo.geocaching/cgeo.geocaching.MainActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
```

After reducing this to "3.0" for image scaling (the two other values remained at "1.3") it crashed at first with the following error, but recovered after that and continued fine after that. (Not sure where the exact break in the logfile should be. The following lines do follow the first log extract from above immediately.)

```
2022-05-01 20:02:33.934 1543-1666/system_process E/InputDispatcher: channel '2885e25f cgeo.geocaching/cgeo.geocaching.MainActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2022-05-01 20:02:33.934 1543-1666/system_process E/InputDispatcher: channel 'e40b877 cgeo.geocaching/cgeo.geocaching.maps.mapsforge.v6.NewMap (server)' ~ Channel is unrecoverably broken and will be disposed!
2022-05-01 20:02:51.475 8453-8515/cgeo.geocaching E/art: Throwing OutOfMemoryError "Failed to allocate a 1102512 byte allocation with 874053 free bytes and 853KB until OOM"
2022-05-01 20:03:14.543 8453-8453/cgeo.geocaching E/art: Throwing OutOfMemoryError "Failed to allocate a 106692 byte allocation with 59028 free bytes and 57KB until OOM"
2022-05-01 20:03:14.578 8453-8453/cgeo.geocaching E/AndroidRuntime: FATAL EXCEPTION: main
    Process: cgeo.geocaching, PID: 8453
    java.lang.RuntimeException: Unable to start activity ComponentInfo{cgeo.geocaching/cgeo.geocaching.maps.mapsforge.v6.NewMap}: android.view.InflateException: Binary XML file line #22: Error inflating class <unknown>
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2298)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
        at android.app.ActivityThread.access$800(ActivityThread.java:144)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:135)
        at android.app.ActivityThread.main(ActivityThread.java:5221)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
     Caused by: android.view.InflateException: Binary XML file line #22: Error inflating class <unknown>
        at android.view.LayoutInflater.createView(LayoutInflater.java:633)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:743)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:806)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:809)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:504)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:414)
        at cgeo.geocaching.databinding.MapMapsforgeV6Binding.inflate(MapMapsforgeV6Binding.java:49)
        at cgeo.geocaching.databinding.MapMapsforgeV6Binding.inflate(MapMapsforgeV6Binding.java:43)
        at cgeo.geocaching.maps.mapsforge.v6.NewMap.onCreate(NewMap.java:271)
        at android.app.Activity.performCreate(Activity.java:5937)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360) 
        at android.app.ActivityThread.access$800(ActivityThread.java:144) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:135) 
        at android.app.ActivityThread.main(ActivityThread.java:5221) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:372) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694) 
     Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
        at android.view.LayoutInflater.createView(LayoutInflater.java:607)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:743) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:504) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:414) 
        at cgeo.geocaching.databinding.MapMapsforgeV6Binding.inflate(MapMapsforgeV6Binding.java:49) 
        at cgeo.geocaching.databinding.MapMapsforgeV6Binding.inflate(MapMapsforgeV6Binding.java:43) 
        at cgeo.geocaching.maps.mapsforge.v6.NewMap.onCreate(NewMap.java:271) 
        at android.app.Activity.performCreate(Activity.java:5937) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360) 
        at android.app.ActivityThread.access$800(ActivityThread.java:144) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:135) 
        at android.app.ActivityThread.main(ActivityThread.java:5221) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:372) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694) 
     Caused by: android.view.InflateException: Binary XML file line #21: Error inflating class <unknown>
        at android.view.LayoutInflater.createView(LayoutInflater.java:633)
        at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:55)
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:682)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:741)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:806)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:479)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:414)
        at android.widget.ZoomControls.<init>(ZoomControls.java:51)
        at android.widget.ZoomControls.<init>(ZoomControls.java:42)
        at org.mapsforge.map.android.input.MapZoomControls.<init>(MapZoomControls.java:138)
        at org.mapsforge.map.android.view.MapView.<init>(MapView.java:159)
        at cgeo.geocaching.maps.mapsforge.v6.MfMapView.<init>(MfMapView.java:31)
        at java.lang.reflect.Constructor.newInstance(Native Method) 
        at java.lang.reflect.Constructor.newInstance(Constructor.java:288) 
        at android.view.LayoutInflater.createView(LayoutInflater.java:607) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:743) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:504) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:414) 
        at cgeo.geocaching.databinding.MapMapsforgeV6Binding.inflate(MapMapsforgeV6Binding.java:49) 
        at cgeo.geocaching.databinding.MapMapsforgeV6Binding.inflate(MapMapsforgeV6Binding.java:43) 
        at cgeo.geocaching.maps.mapsforge.v6.NewMap.onCreate(NewMap.java:271) 
        at android.app.Activity.performCreate(Activity.java:5937) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360) 
        at android.app.ActivityThread.access$800(ActivityThread.java:144) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:135) 
        at android.app.ActivityThread.main(ActivityThread.java:5221) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:372) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694) 
     Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
        at android.view.LayoutInflater.createView(LayoutInflater.java:607)
        at com.android.internal.policy.impl.PhoneLayoutInflater.onCreateView(PhoneLayoutInflater.java:55) 
        at android.view.LayoutInflater.onCreateView(LayoutInflater.java:682) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:741) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:479) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:414) 
        at android.widget.ZoomControls.<init>(ZoomControls.java:51) 
        at android.widget.ZoomControls.<init>(ZoomControls.java:42) 
        at org.mapsforge.map.android.input.MapZoomControls.<init>(MapZoomControls.java:138) 
        at org.mapsforge.map.android.view.MapView.<init>(MapView.java:159) 
        at cgeo.geocaching.maps.mapsforge.v6.MfMapView.<init>(MfMapView.java:31) 
        at java.lang.reflect.Constructor.newInstance(Native Method) 
        at java.lang.reflect.Constructor.newInstance(Constructor.java:288) 
        at android.view.LayoutInflater.createView(LayoutInflater.java:607) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:743) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:809) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:504) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:414) 
        at cgeo.geocaching.databinding.MapMapsforgeV6Binding.inflate(MapMapsforgeV6Binding.java:49) 
        at cgeo.geocaching.databinding.MapMapsforgeV6Binding.inflate(MapMapsforgeV6Binding.java:43) 
        at cgeo.geocaching.maps.mapsforge.v6.NewMap.onCreate(NewMap.java:271) 
        at android.app.Activity.performCreate(Activity.java:5937) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360) 
        at android.app.ActivityThread.access$800(ActivityThread.java:144) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:135) 
        at android.app.ActivityThread.main(ActivityThread.java:5221) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:372) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694) 
     Caused by: java.lang.OutOfMemoryError: Failed to allocate a 106692 byte allocation with 59028 free bytes and 57KB until OOM
        at dalvik.system.VMRuntime.newNonMovableArray(Native Method)
        at android.graphics.BitmapFactory.nativeDecodeAsset(Native Method)
        at android.graphics.BitmapFactory.decodeStream(BitmapFactory.java:609)
        at android.graphics.BitmapFactory.decodeResourceStream(BitmapFactory.java:444)
        at android.graphics.drawable.Drawable.createFromResourceStream(Drawable.java:973)
        at android.content.res.Resources.loadDrawableForCookie(Resources.java:2423)
    	at android.co
2022-05-01 20:03:16.544 1543-1666/system_process E/InputDispatcher: channel '31ac4156 cgeo.geocaching/cgeo.geocaching.MainActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
2022-05-01 20:03:16.544 1543-1666/system_process E/InputDispatcher: channel '3401289 cgeo.geocaching/cgeo.geocaching.MainActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
```

Not sure if it's related, posted it here for reference if it happens to occur again.

--- a/.idea/checkstyle-idea.xml
+++ b/.idea/checkstyle-idea.xml
@@ -3,12 +3,12 @@
   <component name="CheckStyle-IDEA">
     <option name="configuration">
       <map>
-        <entry key="active-configuration" value="PROJECT_RELATIVE:$PROJECT_DIR$/checkstyle.xml:cgeo" />
+        <entry key="active-configuration-0" value="PROJECT_RELATIVE:C:/private/geocaching/github/cgeo/checkstyle.xml:cgeo" />
         <entry key="checkstyle-version" value="8.44" />
         <entry key="copy-libs" value="false" />
-        <entry key="location-0" value="BUNDLED:(bundled):Sun Checks" />
-        <entry key="location-1" value="BUNDLED:(bundled):Google Checks" />
-        <entry key="location-2" value="PROJECT_RELATIVE:$PROJECT_DIR$/checkstyle.xml:cgeo" />
+        <entry key="location-0" value="BUNDLED:(bundled):Sun Checks;All" />
+        <entry key="location-1" value="BUNDLED:(bundled):Google Checks;All" />
+        <entry key="location-2" value="PROJECT_RELATIVE:$PROJECT_DIR$/checkstyle.xml:cgeo;All" />
         <entry key="scan-before-checkin" value="true" />
         <entry key="scanscope" value="JavaOnlyWithTests" />
         <entry key="suppress-errors" value="false" />

--- a/main/res/values/strings.xml
+++ b/main/res/values/strings.xml
@@ -2450,6 +2450,12 @@
     <string name="mapserver_hylly_themes_info">Detailed map theme, built for \"Hylly\" maps</string>
     <string name="brouter_info">Offline routing service</string>
     <string name="maptheme_elevate_categoryinfo">Relevant for: [A] Areas, [P] POI, [R] Routes, [W] Ways</string>
+    <string name="maptheme_scale_map_title">Map Scale Factor</string>
+    <string name="maptheme_scale_map_summary">Scales map; 100% = no scaling</string>
+    <string name="maptheme_scale_text_title">Text Scale factor</string>
+    <string name="maptheme_scale_text_summary">Additionally scales text labels in map; 100% = no extra scaling</string>
+    <string name="maptheme_scale_symbol_title">Symbol Scale factor</string>
+    <string name="maptheme_scale_symbol_summary">Additionally scales symbol sizes in map; 100% = no extra scaling</string>
 
     <!-- history track -->
     <string name="map_clear_trailhistory">Clear history track</string>

--- a/main/src/cgeo/geocaching/maps/mapsforge/v6/NewMap.java
+++ b/main/src/cgeo/geocaching/maps/mapsforge/v6/NewMap.java
@@ -418,7 +418,7 @@ public class NewMap extends AbstractBottomNavigationActivity implements Observer
 
             final boolean tileLayerHasThemes = tileLayerHasThemes();
             menu.findItem(R.id.menu_theme_mode).setVisible(tileLayerHasThemes);
-            menu.findItem(R.id.menu_theme_options).setVisible(tileLayerHasThemes && this.renderThemeHelper.themeOptionsAvailable());
+            menu.findItem(R.id.menu_theme_options).setVisible(tileLayerHasThemes);
             menu.findItem(R.id.menu_theme_legend).setVisible(tileLayerHasThemes && RenderThemeLegend.supportsLegend());
 
             menu.findItem(R.id.menu_as_list).setVisible(!caches.isDownloading() && caches.getVisibleCachesCount() > 1);

--- a/main/src/cgeo/geocaching/maps/mapsforge/v6/RenderThemeHelper.java
+++ b/main/src/cgeo/geocaching/maps/mapsforge/v6/RenderThemeHelper.java
@@ -96,6 +96,7 @@ public class RenderThemeHelper implements XmlRenderThemeMenuCallback {
     //current Theme style menu settings
     private XmlRenderThemeStyleMenu themeStyleMenu;
     private String prefThemeOptionMapStyle = "";
+    private String prefThemeStyleKey = "";
 
     //the last used Zip Resource Provider is cached.
     private static String cachedZipProviderFilename = null;
@@ -157,8 +158,9 @@ public class RenderThemeHelper implements XmlRenderThemeMenuCallback {
         //try to apply stored value
         ThemeData selectedTheme = setSelectedMapThemeInternal(Settings.getSelectedMapRenderTheme());
 
+
         if (selectedTheme == null) {
-            rendererLayer.setXmlRenderTheme(InternalRenderTheme.OSMARENDER);
+            applyDefaultTheme(rendererLayer);
         } else {
             try {
                 //get the theme
@@ -167,6 +169,8 @@ public class RenderThemeHelper implements XmlRenderThemeMenuCallback {
                 // Validate the theme
                 org.mapsforge.map.rendertheme.rule.RenderThemeHandler.getRenderTheme(AndroidGraphicFactory.INSTANCE, new DisplayModel(), xmlRenderTheme);
                 rendererLayer.setXmlRenderTheme(xmlRenderTheme);
+                //setting xmlrendertheme has filled prefThemeStyleKey -> now apply scales
+                applyScales(rendererLayer, prefThemeStyleKey);
             } catch (final IOException e) {
                 Log.w("Failed to set render theme", e);
                 ActivityMixin.showApplicationToast(LocalizationUtils.getString(R.string.err_rendertheme_file_unreadable));
@@ -186,6 +190,23 @@ public class RenderThemeHelper implements XmlRenderThemeMenuCallback {
             tileCache.purge();
         }
         rendererLayer.requestRedraw();
+    }
+
+    private void applyDefaultTheme(final TileRendererLayer rendererLayer) {
+        rendererLayer.setXmlRenderTheme(InternalRenderTheme.OSMARENDER);
+        applyScales(rendererLayer, Settings.RENDERTHEMESCALE_DEFAULTKEY);
+    }
+
+    private void applyScales(final TileRendererLayer rendererLayer, final String themeStyleId) {
+
+        final int mapScale = Settings.getMapRenderScale(themeStyleId, Settings.RenderThemeScaleType.MAP);
+        final int textScale = Settings.getMapRenderScale(themeStyleId, Settings.RenderThemeScaleType.TEXT);
+        final int symbolScale = Settings.getMapRenderScale(themeStyleId, Settings.RenderThemeScaleType.SYMBOL);
+
+        rendererLayer.getDisplayModel().setUserScaleFactor(mapScale / 100f);
+        rendererLayer.setTextScale(textScale / 100f);
+        DisplayModel.symbolScale = symbolScale / 100f;
+
     }
 
     private XmlRenderTheme createThemeFor(@NonNull final ThemeData theme) throws IOException {
@@ -272,7 +293,7 @@ public class RenderThemeHelper implements XmlRenderThemeMenuCallback {
     public void selectMapThemeOptions() {
         final Intent intent = new Intent(activity, RenderThemeSettings.class);
         intent.setFlags(Intent.FLAG_ACTIVITY_NEW_DOCUMENT);
-        if (themeStyleMenu != null) {
+        if (themeOptionsAvailable() && themeStyleMenu != null) {
             intent.putExtra(RenderThemeSettingsFragment.RENDERTHEME_MENU, themeStyleMenu);
         }
         activity.startActivity(intent);
@@ -295,7 +316,7 @@ public class RenderThemeHelper implements XmlRenderThemeMenuCallback {
         themeStyleMenu = menu;
         prefThemeOptionMapStyle = menu.getId();
         final String id = this.sharedPreferences.getString(themeStyleMenu.getId(), themeStyleMenu.getDefaultValue());
-
+        prefThemeStyleKey = menu.getId() + "-" + id;
         final XmlRenderThemeStyleLayer baseLayer = themeStyleMenu.getLayer(id);
         if (baseLayer == null) {
             Log.w("Invalid style " + id);
@@ -316,9 +337,6 @@ public class RenderThemeHelper implements XmlRenderThemeMenuCallback {
     /**
      * Set a new map theme. The theme is evaluated against available themes and possibly corrected.
      * Next time a map viewer is opened, the theme will be evaluated and used if possible
-     *
-     * @param namTheme theme to set
-     * @param theme really set (might be corrected if given value is not correct/incomplete)
      */
     public static boolean setSelectedMapThemeDirect(final String themeIdCandidate) {
         return setSelectedMapThemeInternal(themeIdCandidate) != null;

--- a/main/src/cgeo/geocaching/maps/mapsforge/v6/RenderThemeSettingsFragment.java
+++ b/main/src/cgeo/geocaching/maps/mapsforge/v6/RenderThemeSettingsFragment.java
@@ -1,16 +1,22 @@
 package cgeo.geocaching.maps.mapsforge.v6;
 
 import cgeo.geocaching.R;
+import cgeo.geocaching.settings.SeekbarPreference;
 import cgeo.geocaching.settings.Settings;
 
 import android.app.Activity;
+import android.content.Context;
 import android.os.Bundle;
 
+import androidx.annotation.StringRes;
 import androidx.preference.CheckBoxPreference;
 import androidx.preference.ListPreference;
 import androidx.preference.Preference;
 import androidx.preference.PreferenceCategory;
 import androidx.preference.PreferenceFragmentCompat;
+import androidx.preference.PreferenceGroup;
+import androidx.preference.PreferenceManager;
+import androidx.preference.PreferenceViewHolder;
 
 import java.util.Locale;
 import java.util.Map;
@@ -22,6 +28,7 @@ public class RenderThemeSettingsFragment extends PreferenceFragmentCompat {
     public static final String RENDERTHEME_MENU = "renderthememenu";
 
     ListPreference baseLayerPreference;
+
     XmlRenderThemeStyleMenu renderthemeOptions;
     PreferenceCategory renderthemeMenu;
 
@@ -31,11 +38,9 @@ public class RenderThemeSettingsFragment extends PreferenceFragmentCompat {
 
         // if the render theme has a style menu, its data is delivered via the intent
         renderthemeOptions = (XmlRenderThemeStyleMenu) getActivity().getIntent().getSerializableExtra(RENDERTHEME_MENU);
-        if (renderthemeOptions != null) {
-            // the preference category serves as the hook to add a list preference to allow users to select a style
-            this.renderthemeMenu = (PreferenceCategory) findPreference(getString(R.string.pref_theme_menu));
-            createRenderthemeMenu();
-        }
+        // the preference category serves as the hook to add a list preference to allow users to select a style
+        this.renderthemeMenu = (PreferenceCategory) findPreference(getString(R.string.pref_theme_menu));
+        createRenderthemeMenu();
     }
 
     @Override
@@ -49,12 +54,33 @@ public class RenderThemeSettingsFragment extends PreferenceFragmentCompat {
 
         this.renderthemeMenu.removeAll();
 
+        final String themeStylePrefKey;
+        if (renderthemeOptions != null) {
+            final String themePrefKey = createThemePreferences(activity);
+            themeStylePrefKey = themePrefKey + "-" + PreferenceManager.getDefaultSharedPreferences(activity).getString(themePrefKey, "none");
+        } else {
+            themeStylePrefKey = Settings.RENDERTHEMESCALE_DEFAULTKEY;
+        }
+
+        //scale preferences for theme
+        addScalePreference(activity, renderthemeMenu, Settings.getMapRenderScalePreferenceKey(themeStylePrefKey, Settings.RenderThemeScaleType.MAP),
+            R.string.maptheme_scale_map_title, R.string.maptheme_scale_map_summary);
+        addScalePreference(activity, renderthemeMenu, Settings.getMapRenderScalePreferenceKey(themeStylePrefKey, Settings.RenderThemeScaleType.TEXT),
+            R.string.maptheme_scale_text_title, R.string.maptheme_scale_text_summary);
+        addScalePreference(activity, renderthemeMenu, Settings.getMapRenderScalePreferenceKey(themeStylePrefKey, Settings.RenderThemeScaleType.SYMBOL),
+            R.string.maptheme_scale_symbol_title, R.string.maptheme_scale_symbol_summary);
+
+    }
+
+    private String createThemePreferences(final Activity activity) {
         this.baseLayerPreference = new ListPreference(activity);
         baseLayerPreference.setTitle(R.string.settings_title_map_style);
 
         // the id of the setting is the id of the stylemenu, that allows this
         // app to store different settings for different render themes.
-        baseLayerPreference.setKey(this.renderthemeOptions.getId());
+        final String themePrefKey = this.renderthemeOptions.getId();
+        baseLayerPreference.setKey(themePrefKey);
+
 
         // this is the user language for the app, in 'en', 'de' etc format
         // no dialects are supported at the moment
@@ -96,6 +122,7 @@ public class RenderThemeSettingsFragment extends PreferenceFragmentCompat {
         });
         renderthemeMenu.addPreference(baseLayerPreference);
 
+
         // additional theme info
         if (RenderThemeHelper.getRenderThemeType() == RenderThemeHelper.RenderThemeType.RTT_ELEVATE) {
             final Preference info = new Preference(activity);
@@ -124,6 +151,28 @@ public class RenderThemeSettingsFragment extends PreferenceFragmentCompat {
             checkbox.setIconSpaceReserved(false);
             this.renderthemeMenu.addPreference(checkbox);
         }
+        return themePrefKey;
+    }
+
+    private void addScalePreference(final Context context, final PreferenceGroup cat, final String prefKey, @StringRes final int titleId, @StringRes final int summaryId) {
+
+        final Preference info = new Preference(context) {
+            public void onBindViewHolder(final PreferenceViewHolder holder) {
+                super.onBindViewHolder(holder);
+                holder.setDividerAllowedAbove(false);
+                holder.setDividerAllowedBelow(false);
+            }
+        };
+        info.setTitle(titleId);
+        info.setSummary(summaryId);
+        info.setIconSpaceReserved(false);
+        cat.addPreference(info);
+
+        final SeekbarPreference seek = new SeekbarPreference(context, 10, 500, "", "%",
+            new SeekbarPreference.FactorizeValueMapper(10));
+        seek.setDefaultValue(100);
+        seek.setKey(prefKey);
+        cat.addPreference(seek);
     }
 
 }

--- a/main/src/cgeo/geocaching/settings/SeekbarPreference.java
+++ b/main/src/cgeo/geocaching/settings/SeekbarPreference.java
@@ -30,7 +30,35 @@ public class SeekbarPreference extends Preference {
     protected final int defaultValue = 10;
     protected boolean hasDecimals = false;
     protected String label = "";
+    protected String unitValue = "";
     protected Context context = null;
+    private ValueProgressMapper valueProgressMapper = null;
+
+    public interface ValueProgressMapper {
+        int valueToProgress(int value);
+
+        int progressToValue(int progress);
+
+    }
+
+    public static class FactorizeValueMapper implements ValueProgressMapper {
+        private final int factor;
+
+        public FactorizeValueMapper(final int factor) {
+            this.factor = factor;
+        }
+
+
+        @Override
+        public int valueToProgress(final int value) {
+            return value / factor;
+        }
+
+        @Override
+        public int progressToValue(final int progress) {
+            return progress * factor;
+        }
+    }
 
     public SeekbarPreference(final Context context) {
         this(context, null);
@@ -43,6 +71,22 @@ public class SeekbarPreference extends Preference {
     public SeekbarPreference(final Context context, final AttributeSet attrs, final int defStyle) {
         super(context, attrs, defStyle);
         this.context = context;
+        initInternal(attrs);
+    }
+
+    /** for programmatic creation */
+    public SeekbarPreference(final Context context, final int min, final int max, final String label, final String unitValue, final ValueProgressMapper valueProgressMapper) {
+        super(context, null, android.R.attr.preferenceStyle);
+        this.context = context;
+        this.minProgress = min;
+        this.maxProgress = max;
+        this.label = label == null ? "" : label;
+        this.unitValue = unitValue == null ? "" : unitValue;
+        this.valueProgressMapper = valueProgressMapper;
+        initInternal(null);
+    }
+
+    private void initInternal(final AttributeSet attrs) {
         setPersistent(true);
         setLayoutResource(R.layout.preference_seekbar);
 
@@ -70,11 +114,11 @@ public class SeekbarPreference extends Preference {
     }
 
     protected int valueToProgress(final int value) {
-        return valueToProgressHelper(value);
+        return valueProgressMapper == null ? valueToProgressHelper(value) : valueProgressMapper.valueToProgress(value);
     }
 
     protected int progressToValue(final int progress) {
-        return progress;
+        return valueProgressMapper == null ? progress : valueProgressMapper.progressToValue(progress);
     }
 
     protected String valueToShownValue(final int value) {
@@ -99,7 +143,7 @@ public class SeekbarPreference extends Preference {
      * @return string for the unit label
      */
     protected String getUnitString() {
-        return "";
+        return unitValue;
     }
 
     /**

--- a/main/src/cgeo/geocaching/settings/Settings.java
+++ b/main/src/cgeo/geocaching/settings/Settings.java
@@ -122,6 +122,8 @@ public class Settings {
     private static MapSource mapSource;
     private static AbstractTileProvider tileProvider;
 
+    public static final String RENDERTHEMESCALE_DEFAULTKEY = "renderthemescale_default";
+
     public enum CoordInputFormatEnum {
         Plain,
         Deg,
@@ -229,6 +231,8 @@ public class Settings {
             return title;
         }
     }
+
+    public enum RenderThemeScaleType { MAP, TEXT, SYMBOL }
 
     //NO_APPLICATION_MODE will be true if Settings is used in context of local unit tests
     private static final boolean NO_APPLICATION_MODE = CgeoApplication.getInstance() == null;
@@ -492,7 +496,11 @@ public class Settings {
     }
 
     private static int getInt(final int prefKeyId, final int defaultValue) {
-        return sharedPrefs == null ? defaultValue : sharedPrefs.getInt(getKey(prefKeyId), defaultValue);
+        return getIntDirect(getKey(prefKeyId), defaultValue);
+    }
+
+    private static int getIntDirect(final String prefKey, final int defaultValue) {
+        return sharedPrefs == null ? defaultValue : sharedPrefs.getInt(prefKey, defaultValue);
     }
 
     public static int getPreferencesCount() {
@@ -1591,6 +1599,17 @@ public class Settings {
     /** Shall SOLELY be used by {@link cgeo.geocaching.maps.mapsforge.v6.RenderThemeHelper}! */
     public static void setSyncMapRenderThemeFolder(final boolean syncMapRenderThemeFolder) {
         putBoolean(R.string.pref_renderthemefolder_synctolocal, syncMapRenderThemeFolder);
+    }
+
+    /** Shall SOLELY be used by {@link cgeo.geocaching.maps.mapsforge.v6.RenderThemeSettingsFragment}! */
+    public static String getMapRenderScalePreferenceKey(final String themeStyleId, final RenderThemeScaleType scaleType) {
+        return themeStyleId + "-" + scaleType;
+    }
+
+    /** Shall SOLELY be used by {@link cgeo.geocaching.maps.mapsforge.v6.RenderThemeHelper}! */
+    public static int getMapRenderScale(final String themeStyleId, final RenderThemeScaleType scaleType) {
+        final int value = getIntDirect(getMapRenderScalePreferenceKey(themeStyleId, scaleType), 100);
+        return Math.min(500, Math.max(value, 10));
     }
 
     /**

